lib/services/gitlab_issues.rb

```diff
@@ -0,0 +1,323 @@
+        ['Feature -> Issue, Requirement -> Checklist item', 'issue-checklist']
+    ], description: 'Choose how features and requirements in Aha! will map to issues and checklists in GitLab.'
+
+    string :due_date_phase, description: 'The name of the phase to find the due date of a release', label: 'Due Date Phase'
+    # it looks like the GitHub status_mapping is not available to us, right now we'll need to assume the statuses
+    # that we want: Open -> Ready To Develop, closed -> Ready to Ship
+    # internal :status_mapping
+
+    boolean :add_status_labels, description: 'Sync the Aha! status using a label on the GitLab issue'
+
+    callback_url description: 'Use this URL to setup a two-way integration with GitLab issues.'
+
+    def receive_installed
+        meta_data.repos = repo_resource.all.map { |repo| { full_name: repo['name'], id: repo['id'], web_url: repo['web_url']  } }
+    end
+
+    def server_url
+        if data.server_url.present?
+            data.server_url
+        else
+            'https://www.gitlab.com/api/v3/'
+        end
+    end
+
+    def receive_create_feature
+      milestone = find_or_attach_gitlab_milestone(payload.feature.release)
+      issue = find_or_attach_gitlab_issue(payload.feature, milestone)
+      update_requirements(payload.feature.requirements, milestone, issue["id"])
+    end
+
+    def receive_create_release
+      find_or_attach_gitlab_milestone(payload.release)
+    end
+
+    def receive_update_feature
+      milestone = find_or_attach_gitlab_milestone(payload.feature.release)
+      issue = update_or_attach_gitlab_issue(payload.feature, milestone)
+      update_requirements(payload.feature.requirements, milestone, issue["id"])
+    end
+
+    def receive_update_release
+      update_or_attach_gitlab_milestone(payload.release)
+    end
+
+    def add_status_labels_enabled?
+      if data.add_status_labels.is_a?(TrueClass) || data.add_status_labels.is_a?(FalseClass)
+        data.add_status_labels
+      elsif data.add_status_labels.is_a?(String)
+        !data.add_status_labels.to_i.zero?
+      elsif data.add_status_labels.is_a?(Numeric)
+        !data.add_status_labels.zero?
+      else
+        false
+      end
+    end
+
+    def receive_webhook
+      #action = payload.webhook.action
+      #issue = payload.webhook.issue
+      objattr = payload.webhook.object_attributes
+      if objattr
+        action = objattr.action
+      end
+      return unless objattr and action
+      results = api.search_integration_fields(data.integration_id, "number", objattr.id)['records'] rescue []
+      return unless results.size == 1
+      if resource = results[0].requirement then
+        resource_kind = :requirement
+      elsif resource = results[0].feature then
+        resource_kind = :feature
+      else
+        return
+      end
+
+      diff = {}
+      diff[:name] = objattr.title if resource.name != objattr.title
+
+      case action
```

Comment:
Ah, yeah, reading the docs I see what you mean. If you don't mind working on it, it would be nice for the GitLab integration to have parity with GitHub; I'll need to discuss internally and see how much of a priority that is. But if we don't end up implementing this, we should remove the "Status label mapping" configuration/option entirely. It was mainly designed so that the developers could update the status label in GitHub and then the integration would update the Aha! status so the PM can see progress; I don't think it makes any sense to have it without the two-way integration.